app/grpc.go

```diff
@@ -0,0 +1,19 @@
+package app
+
+import (
+	example "golang_template/grpc/gen/example/proto"
+	"golang_template/internal/services"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+)
+
+func (a *application) InitGRPCServer() *grpc.Server {
+	grpcServer := grpc.NewServer()
+	exampleService := services.NewExampleService()
+	example.RegisterExampleServer(grpcServer, exampleService)
```

Comment:
It's better to use the `grpcController` for registration. To execute the code, send the DTOs from the controller to the service layer.

For Example: (pseudo code)
```
// grpcController

func (c *grpcController) SayHello(ctx context.Context, req *example.HelloRequest) (*example.HelloReply, error) {
   data := req.CastToDto()
   return c.grpcService.SeyHello(ctx, data)
}


// dto:
type HelloRequest struct {

}

func (*example.HelloRequest) CastToDto() *HelloRequest{
   .....
}

// grpcService:
func (c *grpcService) SayHello(ctx context.Context, req *dto.HelloRequest) (*example.HelloReply, error) {
   ....
}
```